File: 2021/day01B.py
with open('day01_input.txt') as f:
    nums = [int(x.strip()) for x in f]
    

count = 0

for i in range(len(nums) - 3):
    curr = sum(nums[i:i+3])
    nxt = sum(nums[i+1:i+4])
    count += 1 if nxt > curr else 0

print(f'Part B:\t{count}')

File: 2021/day01B.js
const fs = require('fs')
const filename = 'day01_input.txt'
const encoding = 'utf8'

fs.readFile(filename, encoding, (err, data) => {
    if (err) throw err
    
    const nums = data.split('\n').map(n => Number.parseInt(n, 10))
    const len = nums.length
    const count = nums.reduce((count, curr, i) => {
        if (i + 3 < len) {
            const currSum = curr + nums[i + 1] + nums[i + 2]
            const nextSum = nums[i + 1] + nums[i + 2] + nums[i + 3]

            count = nextSum > currSum ? count + 1 : count
        }

        return count
    }, 0)

    console.log(count)
})
